File: 4B_Before_an_examp.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

def main():
    total_days , hours = [int(i) for i in input().split() ]
    min_hour = []
    max_hour = []

    for i in range(total_days):
        x, y = [int(x) for x in input().split()]
        min_hour.append(x)
        max_hour.append(y)
    if sum(max_hour ) < hours or sum(min_hour) > hours:
        print("NO")
        exit()
    hours -= sum(min_hour)
    for i in range(total_days):
        add = min(max_hour[i]-min_hour[i],hours)
        min_hour[i] = str(min_hour[i]+add)
        hours -= add
    print("YES")
    print(" ".join(min_hour))


if __name__ == "__main__":
    main()

File: 4B_Before_an_examp.cpp
#include <iostream>
using namespace std;
int d,sumTime,a[100],b[100],k,x;
int add;
int main()
{
	cin>>d>>sumTime;
	for(int i = 0; i< d; i++)
	{
		cin>>a[i]>>b[i];
		k+=a[i];
		x+=b[i];
	}
	if (k > sumTime || x < sumTime)
	{
		cout <<"NO"<<endl;
		return 0;
	}

	sumTime -= k;
	int add;
	cout <<"YES"<<endl;
	for (int i = 0; i < d; i++)
	{
		add = min(b[i]-a[i],sumTime);
	 cout << a[i]+add<<" ";
	 sumTime -= add;
	}
	

	return 0;
}
